### ISSUE TITLE
[Bug] Transform throws error when schema uses a nested projection

### ISSUE THREAD

**Issue by C-Higgins** [nobrainr/axios-morphism#15]
As seen [here](https://github.com/nobrainr/morphism/issues/147), it should be possible to create a nested schema like so:

```typescript
const schema = createSchema<target, source>({
  foo: {
    id: 'id'
  }
})
```

This works when applying the schema with a simple `morphism` call. However, if you pass this schema to the Axios adapter, it will fail and throw an error.

**Comment by C-Higgins** [comment]
It looks like this might be because the morphism version is out of date in the dist. Though npm reports both morphism and the morphism dependency here as 1.12.2, if you look at the actual code it is not the same.

The dist code in question:
```js
if (o.isObject(l)) {
                var h = void 0;
                try {
                  var v = void 0;
                  Array.isArray(l.path) ? (v = o.aggregator(l.path, e)) : o.isString(l.path) && (v = o.get(e, l.path)),
                    (h = l.fn.call(void 0, v, e, t, i));
                } catch (e) {
                  throw ((e.message =
                    'Unable to set target property [' +
                    p +
                    '].\n                                \n An error occured when applying [' +
                    l.fn.name +
                    '] on property [' +
                    l.path +
                    ']\n                                \n Internal error: ' +
                    e.message),
                  e);
                }
                return ((s = {})[p] = h), s;
              }
```

This is different than the code in morphism, where it does a check for the object properties:
```js
if (a(t))
            return (e) => {
              var n,
                {object: o, items: i, objectToCompute: a} = e;
              try {
                var c;
                Array.isArray(t.path) ? (c = p(t.path, o)) : l(t.path) && (c = g(o, t.path)),
                  (n = t.fn.call(void 0, c, o, i, a));
              } catch (e) {
                throw ((e.message = 'Unable to set target property ['
                  .concat(r, '].\n                                        \n An error occured when applying [')
                  .concat(t.fn.name, '] on property [')
                  .concat(t.path, ']\n                                        \n Internal error: ')
                  .concat(e.message)),
                e);
              }
              return n;
            };
```
Here, a(t) is the function 
```js
function a(e) {
        return h(e) && e.hasOwnProperty('fn') && e.hasOwnProperty('path');
      }
```

**Comment by emyann** [comment]
@C-Higgins Thank you for reporting! You're absolutely right, this is exactly the issue. I think this is a mistake that I added `morphism` as a dependency, it should be a peer dependency instead as `axios` is. 

I'm setting up `Greenkeeper` on this repo so that it will keep the dependencies up to date and I'll update this library so that `Morphism` will be a peer dependency.

**Comment by emyann** [comment]
@C-Higgins The issue has been fixed with version [`1.1.0`](https://github.com/nobrainr/axios-morphism/releases/tag/v1.1.0). I'll proceed with the subsequent update I mentioned above.

Thank you again for having filled this issue 🙏🏽

**Comment by C-Higgins** [comment]
Thanks for the quick response time!
